File: DataGridView buttons (C#)/[C#]-DataGridView buttons (C#)/C#/TeamBaseLibrary/BaseExceptionsHandler.cs
using System;

namespace TeamBaseLibrary
{
    public class BaseExceptionsHandler
    {

        protected bool mHasException;
        /// <summary>
        /// Indicate the last operation thrown an 
        /// exception or not
        /// </summary>
        /// <returns></returns>
        public bool HasException
        {
            get
            {
                return mHasException;
            }
        }
        protected Exception mLastException;
        /// <summary>
        /// Provides access to the last exception thrown
        /// </summary>
        /// <returns></returns>
        public Exception LastException
        {
            get
            {
                return mLastException;
            }
        }
        /// <summary>
        /// If you don't need the entire exception as in 
        /// LastException this provides just the text of the exception
        /// </summary>
        /// <returns></returns>
        public string LastExceptionMessage
        {
            get
            {
                return mLastException.Message;
            }
        }
        /// <summary>
        /// Indicate for return of a function if there was an 
        /// exception thrown or not.
        /// </summary>
        /// <returns></returns>
        public bool IsSuccessFul
        {
            get
            {
                return !mHasException;
            }
        }
    }
}

File: Efficiently working with data readers for SQL-Server in VB.NET/[VB]-Efficiently working with data readers for SQL-Server in VB.NET/VB/WindowsApp1/Classes/BaseExceptionsHandler.vb
Imports System.Data.SqlClient

Public Class BaseExceptionsHandler

    Protected mHasException As Boolean
    Public ReadOnly Property HasException() As Boolean
        Get
            Return mHasException
        End Get
    End Property
    Protected mHasSqlException As Boolean
    Public ReadOnly Property HasSqlException() As Boolean
        Get
            Return mHasSqlException
        End Get
    End Property
    Protected mSqlException As SqlException
    Public ReadOnly Property LastSqlException() As SqlException
        Get
            Return mSqlException
        End Get
    End Property
    Protected mLastException As Exception
    Public ReadOnly Property LastException() As Exception
        Get
            Return mLastException
        End Get
    End Property
    Public ReadOnly Property LastExceptionMessage As String
        Get
            Return mLastException.Message
        End Get
    End Property
    ''' <summary>
    ''' Indicate for return of a function if there was an 
    ''' exception thrown or not.
    ''' </summary>
    ''' <returns></returns>
    Public ReadOnly Property IsSuccessFul As Boolean
        Get
            Return Not mHasException OrElse Not mHasSqlException
        End Get
    End Property
End Class
